app/assets/javascripts/discourse/app/form-kit/components/fk/form.gjs

```diff
@@ -34,11 +35,19 @@ class FKForm extends Component {
   constructor() {
     super(...arguments);
 
-    this.args.onRegisterApi?.({
-      set: this.set,
-      setProperties: this.setProperties,
-      submit: this.onSubmit,
-      reset: this.onReset,
+    const isDirtyCheck = () => {
+      return this.formData.isDirty;
+    };
+    next(() => {
```

Comment:
It's not needed in tests, but on a "real" page we will have an exception without the next.

Reply:
what's the exception?

Reply:
I know this is just an example, but so we need to support using `isDirty` in a template **outside the form**?

Ideally `<Form>` would expose it as a block parameter (either isDirty directly, or through `api` object) and that would avoid this issue altogether.

Reply:
yes sometimes we will have other side effects (like showing another component outside the form) depending on the state of the form